projects/table-builder/src/lib/classes/TableTemplateBuilder.ts

```diff
@@ -27,10 +27,10 @@ export class TableTemplateBuilder {
       };
     }
 
-    getColumnNames(): Observable<string[]> {
+    getColumnNames(): Observable<MetaData[]> {
```

Comment:
I think the name of the method needs to change to make it current and accurate.

Reply:
What should we call it?
getMetaData?
The _emes_ is, I don't need all the meta - just the key and the display name, but I didn't want to make a new type, or start strongly-typing `{ key: string, displayName: string }` all over the place, so I just passed along the whole meta data array.
Thoughts?

Reply:
It appears that this is getting all non hidden columns.
While I'm looking at it I'm wondering.
We should probably not need to look at the templates to get the meta data.
Can we try to get the MetaData directly from the able builder?

Reply:
I updated the method.  As a side note, what is the purpose of `FieldType.Hidden`?  Omitting the one meta data object altogether has the same effect as including it and setting its fieldType to FieldType.Hidden.  Under what circumstances would you choose the latter over the former?